### PR TITLE
Development Environment Setup

### PR DESCRIPTION
**Pull request by gmacgre**
Set up the necessary items for the development of the bot, including:
TypeScript
node modules
ESLint
Prettier
Prisma
SQLite
dotenv
changelog
and a basic README

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Environment variables declared in this file are automatically made available to Prisma.
```
Comment (AverageHelper) [inline .env:1]: Please add .env to the .gitignore file

**Review by AverageHelper — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+node_modules
+node_modules/*
+dist
+dist/*
```
Comment (SpencerHastings) [inline .gitignore:4]: I don't think you need to duplicate these.

Comment (JstnMcBrd) [reply .gitignore:4]: I'll fix that too.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        "url": "git+https://github.com/BYU-CS-Discord/CSBot.git"
+    },
+    "author": "",
+    "license": "Unlicense",
+    "bugs": {
+        "url": "https://github.com/BYU-CS-Discord/CSBot/issues"
+    },
+    "homepage": "https://github.com/BYU-CS-Discord/CSBot#readme",
+    "devDependencies": {
+        "@types/sqlite3": "^3.1.8",
```
Comment (SpencerHastings) [inline package.json:20]: Not sure, but figured I'd check, are these sqlite types necessary?

File: .env (hunk 1 above)
Comment (JstnMcBrd) [reply .env]: Done.

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Set up an environment for the project.
```
Comment (SpencerHastings) [inline CHANGELOG.md:1]: Something like this format might be nice https://keepachangelog.com/en/1.0.0/

File: package.json (hunk 3 above)
Comment (AverageHelper) [reply package.json:20]: They aren't for Prisma

Comment (SpencerHastings) [reply package.json:20]: Oh there's a sqlite3 dependency below, is that necessary?

Comment (JstnMcBrd) [reply package.json:20]: Should I remove that line?

Comment (SpencerHastings) [reply package.json:20]: If prisma takes care of the database connection I don't think sqlite is a necessary dependency, even if you use sqlite as the backend, idk for sure tho

Comment (JstnMcBrd) [reply package.json:20]: We'll get rid of it for now. Can always add it back later.

File: CHANGELOG.md (hunk 4 above)
Comment (gmacgre) [reply CHANGELOG.md]: Done!

File: package.json (hunk 3 above)
Comment (SpencerHastings) [reply package.json:20]: Yeah I'm pretty sure the sqlite dependency isn't necessary to use sqlite with prisma

File: .eslintignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+node_modules
```
Comment (AverageHelper) [inline .eslintignore:1]: Would this make more sense as an `ignorePatterns` array in `.eslintrc` instead? It's only the two lines, and both of these files deal with ESLint configuration.

**Review by AverageHelper — CHANGES_REQUESTED** [review]

File: .eslintrc [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      "@typescript-eslint"
+    ],
+    "extends": [
+      "eslint:recommended",
+      "plugin:@typescript-eslint/eslint-recommended",
+      "plugin:@typescript-eslint/recommended"
+    ],
+    "rules": {
+      "arrow-spacing": ["warn", { "before": true, "after": true }],
+      "brace-style": ["error", "stroustrup", { "allowSingleLine": true }],
```
Comment (AverageHelper) [inline .eslintrc:14]: We gotta bikeshed this. [Docs here.](https://eslint.org/docs/latest/rules/brace-style)

I prefer the default `1tbs` style. Should we take a poll on the others?

File: .eslintignore (hunk 5 above)
Comment (gmacgre) [reply .eslintignore:1]: Done. Also seems that it implicitly ignores node_modules
https://eslint.org/docs/latest/user-guide/configuring/ignoring-code

**Review by AverageHelper — CHANGES_REQUESTED** [review]
I gotta read the rest of this ESLint config, you've changed a lot of defaults lol

File: package.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        "eslint": "^8.24.0",
+        "eslint-config-prettier": "^8.5.0",
+        "eslint-plugin-prettier": "^4.2.1",
+        "prettier": "^2.7.1",
+        "prisma": "^4.3.1",
+        "typescript": "^4.8.3"
+    },
+    "dependencies": {
+        "dotenv": "^16.0.2",
+        "sqlite3": "^5.1.1"
```
Comment (AverageHelper) [inline package.json:31]: Not needed with Prisma

**Review by AverageHelper — CHANGES_REQUESTED** [review]
Ok, I think I pretty much agree with the rest of the ESLint config, apart from the block style noted in my earlier comment.

File: package.json (hunk 7 above)
Comment (gmacgre) [reply package.json]: Done!

**Review by AverageHelper — APPROVED** [review]

**Pull request merged**